MonoGame.Framework/Vector3.cs

```diff
@@ -522,7 +530,7 @@ public bool Equals(Vector3 other)
         /// <returns>Hash code of this <see cref="Vector3"/>.</returns>
         public override int GetHashCode()
         {
-            return (int)(this.X + this.Y + this.Z);
+            return X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode();
```

Comment:
Nothing has been taked. the XNA 
using "return this.X.GetHashCode() + this.Y.GetHashCode() + this.Z.GetHashCode();"
The new one is "return X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode();" See the difference ?